docs/guides/external-mcp-server.md

```diff
@@ -0,0 +1,299 @@
+- Repository management and queries
+- Issue and pull request operations
+- Code search and file operations
+- GitHub Actions workflow management
+
+More details: https://github.com/github/github-mcp-server
+
+## Prerequisites
+
+**Required Setup:** This guide assumes you've already created a local Kind cluster with MCP Gateway installed:
+
+```bash
+make local-env-setup
+```
+
+This command creates a Kind cluster and installs:
+- Istio service mesh
+- Gateway API CRDs
+- MCP Gateway controller and broker
+- Test MCP servers
+
+If you haven't run this yet, do so before proceeding with this guide.
+
+### GitHub Personal Access Token (PAT)
+For this tutorial, you'll need a GitHub Personal Access Token (PAT) with minimal permissions
+
+**For GitHub.com:**
+1. Go to https://github.com/settings/tokens/new
+2. Select only: `read:user` - Read user profile data
+3. Generate the token
+
+**Production Note:** This static PAT approach is for demonstration only. In production, you would use OAuth flows for per-user authentication.
+
+**Set your token as an environment variable:**
+```bash
+export GITHUB_PAT="ghp_YOUR_GITHUB_TOKEN_HERE"
+```
+
+### Network Requirements
+- Egress access from your cluster to `api.githubcopilot.com` (port 443)
+
+## Verify Prerequisites
+
+Before proceeding, verify all dependencies are met:
+
+```bash
+# MCP Gateway is deployed
+kubectl get mcpservers -A
+kubectl get pods -n mcp-system
+
+# verify egress to GitHub's MCP server from a test pod
+kubectl run -it --rm debug --image=nicolaka/netshoot --restart=Never -- \
+  curl -I https://api.githubcopilot.com/mcp/
+
+# Test your GitHub token (outside cluster)
+curl -H "Authorization: Bearer $GITHUB_PAT" \
+  https://api.github.com/user
+```
+
+## Overview
+
+To connect to an external MCP server using Istio and Gateway API, we need:
+1. A `Gateway` listener for the external hostname (`api.githubcopilot.com`)
+2. An `HTTPRoute` that matches the external hostname
+3. An `ExternalName` Service pointing to the external service
+4. A `ServiceEntry` to define the external service in Istio's service registry
+5. A `DestinationRule` for connection policies
+6. An `MCPServer` CR to register this MCP server with the gateway
+7. A `Secret` containing authentication credentials that the broker/router will use to authenticate with the external MCP server
+
+## Step 1: Add External Hostname to Gateway
+
+The `Gateway` needs a listener for the external hostname:
+
+```bash
+# Add the GitHub listener to the Gateway using kubectl patch
+kubectl patch gateway mcp-gateway -n gateway-system --type json -p='[
+  {
+    "op": "add",
+    "path": "/spec/listeners/-",
+    "value": {
+      "name": "github-external",
+      "hostname": "api.githubcopilot.com",
+      "port": 8080,
+      "protocol": "HTTP",
+      "allowedRoutes": {
+        "namespaces": {
+          "from": "All"
+        }
+      }
+    }
+  }
+]'
+```
+
+## Step 2: Create ExternalName Service
```

Comment:
Are the 7 steps here corresponding to the 7 steps above? If so, step 2 is supposed to be creating the HTTPRoute.